opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -106,4 +110,6 @@ def detect(self) -> "Resource":
                 if found_resources:
                     self.gcp_resources = found_resources
                     break
+        if not self.gcp_resources:
+            raise NoGoogleResourcesFound()
```

Comment:
From a debugging perspective, it would be great if we had a parent exception to `raise NoGoogleResourcesFound() from ex`. E.g. if the call to the metadata server timed-out, then the crash would show that.

Reply:
resolved offline. We now log exceptions that were unexpected (network fail, keyerrors).